### ISSUE TITLE
Add Support for PGID and PUID Docker Enviroment Variables

### ISSUE THREAD

**Issue by matthewparkes** [eclipse/mosquitto#2441]
Version: 2.0.14

When persistenting data volumes all files are owned by user 1883

docker-compose.yml
`
services:

  #Eclipse Mosquitto
  mosquitto:
    container_name: eclipse-mosquitto
    image: eclipse-mosquitto:latest
    restart: unless-stopped
    environment:
      - PGID=1000
      - PUID=1000
      - UMASK_SET=000
    deploy:
      resources:
        limits:
          memory: 125M
    ports:
       - "1883:1883"
       - "9001:9001"
    volumes:
      - ./data-mosquitto/config:/mosquitto/config
      - ./data-mosquitto/data:/mosquitto/data
      - ./data-mosquitto/log:/mosquitto/log
`

Can support for PGID and PUID be added so logs are easily viewble on host machine without ROOT?

Thanks

**Comment by thesebastianf** [comment]
I also ask for support of PGID and PUID as Environment variables of docker compose to be actually acknowledged.

**Comment by rursache** [comment]
this should be priority no. 1

**Comment by FerronN** [comment]
Why is this change only done for the local docker file?

**Comment by SgtErnestBilko** [comment]
It would be really useful if this issue were 'sorted'..... please

**Comment by degudejung** [comment]
+1 for me!

**Comment by zeppmg** [comment]
Hello,
It seems to me that the user parameter does the trick :

```
services:
  mosquitto:
    image: eclipse-mosquitto:latest
    container_name: mosquitto
    user: "1000:1000"
    [...]
```

**Comment by lox** [comment]
Setting `user` didn't work for me, files are still owned by 1883. 

Accepting UID/GID/TZ is a really standard thing for docker containers to do, otherwise makes it very hard to map permissions back to the host.

**Comment by lox** [comment]
Looks like it's on the develop branch, might that make it into a release at some point @ralight?

**Comment by ctrl-f5** [comment]
2024 and this issue still persists. running this container on a system you dont have root access on makes the config file uneditable and config/data/log folders unremovable.

**Comment by mhoogenbosch** [comment]
Setting up a new environment and am really surprised this is still not fixed.

**Comment by SgtErnestBilko** [comment]
> Setting up a new environment and am really surprised this is still not fixed.

I agree.  Makes developing my docker stack a PIA, I can't sync to GitHub without excluding Mosquito config files.

**Comment by ltctceplrm** [comment]
> Setting `user` didn't work for me, files are still owned by 1883.

It worked for me but I had to first stop mosquitto, chown the files with my PUID and GUID, update the docker compose and then restart it.

**Comment by march42** [comment]
I submitted a PR for this, currently under review. I have my image based on the `2.1-alpine` image.

Maybe some of you want to give it a try. Thanks in advance.
Use image `marchefter/eclpise-mosquito:appuser` instead of `eclpise-mosquito:latest` like in the example below. 

```docker-compose.yaml
services:

  #Eclipse Mosquitto
  mosquitto:
    image: marchefter/eclpise-mosquito:appuser
    restart: unless-stopped
    environment:
      - PUID=1000
      - PGID=1000
    ports:
       - "1883:1883"
    volumes:
      - ./mosquitto-config:/mosquitto/config
      - ./mosquitto-data:/mosquitto/data
      - ./mosquitto-log:/mosquitto/log
```

**NOTE:** The image, especially script `docker-entrypoint.sh`, won't change your file permissions on the host. You need to take care of this yourself.
